test/regression.jl

```diff
@@ -0,0 +1,63 @@
+using Test, Random, BSON, AdvancedHMC, Distributions, ForwardDiff
+
+is_ef6de39 = isdefined(AdvancedHMC, :EndPointTS)
```

Comment:
A better way of performing this regression test is to check out a pinned version of AHMC, run the tests on CI machines with the pinned version and the current version. Then we can compare results without worrying machine-specific details.

Reply:
Yes I agree. I don't know how to do that though. Can you point me some example CI codes on doing so?

Reply:
[`BenchmarkTools`](https://github.com/JuliaCI/BenchmarkTools.jl) allows users to write tests, and run regression tests automatically. Our use here is a bit special, but the principles are the same I think.

Reply:
Can you point me an example? I only found example where they are looking at regression in time.

Reply:
In my view, no single version of the code should be considered the ground truth. Rather, we can only be confident that features covered by the test suite work correctly. Hence instead of checking out a previous version to test against, it's better to add any missing tests, ensuring they pass on both the old version and the new version.

Reply:
For performance regressions, you can use PkgBenchmark (see e.g. https://github.com/JuliaFolds/FLoops.jl/blob/master/.github/workflows/benchmark.yml), but I don't think that checks the outputs.

Reply:
Ideally, we can keep PRs small, i.e. one PR for a change/feature. For big changes like this PR, we should try to do it in multiple steps. It would make reviewing code much easier, which leads to much quicker merging too.